sentry_sdk/client.py

```diff
@@ -667,7 +675,7 @@ def capture_event(
             envelope = Envelope(headers=headers)
 
             if is_transaction:
-                if profile is not None:
+                if isinstance(profile, Profile):
```

Comment:
`profile` should only ever be getting set to a `Profile`; however, we can only type the profile as `object` because we otherwise have a circular import when building the Sphinx docs.

We check `isinstance` explicitly now to ensure type safety – behavior should however remain the same since we currently only set `profile` to either a `Profile` or `None`.